march_hardware/src/PDOmap.cpp

```diff
@@ -6,129 +6,151 @@
 
 namespace march4cpp
 {
-PDOmap::PDOmap()
-{
-  this->initAllObjects();
-}
+std::unordered_map<IMCObjectName, IMCObject> PDOmap::all_objects = {
+  { IMCObjectName::StatusWord, IMCObject(0x6041, 16) },
+  { IMCObjectName::ActualPosition, IMCObject(0x6064, 32) },
+  { IMCObjectName::MotionErrorRegister, IMCObject(0x2000, 16) },
+  { IMCObjectName::DetailedErrorRegister, IMCObject(0x2002, 16) },
+  { IMCObjectName::DCLinkVoltage, IMCObject(0x2055, 16) },
+  { IMCObjectName::DriveTemperature, IMCObject(0x2058, 16) },
+  { IMCObjectName::ActualTorque, IMCObject(0x6077, 16) },
+  { IMCObjectName::CurrentLimit, IMCObject(0x207F, 16) },
+  { IMCObjectName::MotorPosition, IMCObject(0x2088, 32) },
+  { IMCObjectName::ControlWord, IMCObject(0x6040, 16) },
+  { IMCObjectName::TargetPosition, IMCObject(0x607A, 32) },
+  { IMCObjectName::TargetTorque, IMCObject(0x6071, 16) },
+  { IMCObjectName::QuickStopDeceleration, IMCObject(0x6085, 32) },
+  { IMCObjectName::QuickStopOption, IMCObject(0x605A, 16) }
+};
 
-void PDOmap::addObject(IMCObjectName objectname)
+void PDOmap::add_object(IMCObjectName object_name)
 {
-  if (this->allObjects.count(objectname) != 1)
+  if (PDOmap::all_objects.find(object_name) == PDOmap::all_objects.end())
   {
-    ROS_WARN("IMC object does not exist (yet), or multiple exist");
+    ROS_WARN("IMC does not exists in in initialised objects so can not be added to PDO");
     return;
   }
-  else if (this->PDOObjects.count(objectname) != 0)
+
+  if (this->PDO_objects.count(object_name) != 0)
   {
-    ROS_WARN("IMC object is already added to PDO map");
+    ROS_WARN("IMC object %i is already added to PDO map", object_name);
```

Comment:
You will probably want to print the hex value or the name of the object, right? This would print a (decimal) integer?

Reply:
Ah I see. It's for you to decide as a team since you'll be using it, not me ;)